platform_plugin_aspects/views.py

```diff
@@ -0,0 +1,48 @@
+"""
+Endpoints for the Aspects platform plugin.
+"""
+
+from django.conf import settings
+from django.contrib.auth.decorators import login_required
+from django.core.exceptions import ImproperlyConfigured
+from django.http import JsonResponse
+from django.views.decorators.http import require_POST
+from opaque_keys.edx.keys import CourseKey
+
+from .utils import _, generate_guest_token
+
+
+@require_POST
+@login_required
+def superset_guest_token(request, course_id):
+    """
+    Return a guest token for Superset that the given request.user can use to access the Superset API.
+    """
+    # TODO -- restrict access to course staff? how?
```

Comment:
I think this would do it:

https://github.com/openedx/edx-platform/blob/41953bb3b2019dd651c0cf2be64b005cf0b47a18/openedx/core/lib/api/permissions.py#L58-L73

Reply:
Or using funtions:

https://github.com/openedx/edx-platform/blob/41953bb3b2019dd651c0cf2be64b005cf0b47a18/lms/djangoapps/teams/api.py#L143C8-L146